OpenSim/Sandbox/ExampleHopperDevice/exampleHopperDevice_answers.cpp

```diff
@@ -0,0 +1,304 @@
+Connectors. For more information, please refer to the Component documentation.
+
+This interactive example consists of three steps:
+  Step 1. Build and simulate a single-legged hopping mechanism.
+  Step 2. Build an assistive device and test it on a simple testbed.
+  Step 3. Connect the device to the hopper to increase hop height.
+
+To start working through this example, go to main() at the bottom of this file.
+From there, you will be directed to specific files and methods in this project
+that need to be completed. Now, hop to it! */
+
+#include <OpenSim/OpenSim.h>
+#include "defineDeviceAndController_answers.h"
+#include "helperMethods.h"
+
+static const double SIGNAL_GEN_CONSTANT{ 0.33 };
+static const double REPORTING_INTERVAL{ 0.2 };
+
+static const std::string testbedAttachment1{"/testbed/ground"};
+static const std::string testbedAttachment2{"/testbed/load"};
+
+//TODO: Provide the full path names of the PhysicalOffsetFrames defined on the
+//      hopper for attaching the assistive device. See buildHopperModel.cpp and
+//      showSubcomponentInfo() in helperMethods.h.
+// [Step 3, Task A]
+//static const std::string thighAttachment{"/Dennis/?????"};
+//static const std::string shankAttachment{"/Dennis/?????"};
+static const std::string thighAttachment{"/Dennis/thigh/deviceAttachmentPoint"};
+static const std::string shankAttachment{"/Dennis/shank/deviceAttachmentPoint"};
+
+//TODO: Provide the name of the output corresponding to the hopper's height.
+//      Hint: the hopper's pelvis is attached to ground with a vertical slider
+//      joint. Also see Device::getHeight() in defineDeviceAndController.h.
+// [Step 3, Task B]
+//static const std::string hopperHeightOutput{"/Dennis/?????"};
+static const std::string hopperHeightOutput{"/Dennis/slider/height/value"};
+
+// To assist hopping, we will activate the knee device whenever the vastus
+// muscle is active. To do this, we will need to connect the vastus muscle's
+// "activation" output to the controller's "activation" input.
+static const std::string signalForKneeDevice{"/Dennis/vastus/activation"};
+
+
+namespace OpenSim {
+
+// Forward declarations for methods used below.
+Model buildHopper();    //defined in buildHopperModel.cpp
+Model buildTestbed();   //defined in helperMethods.h
+Device* buildDevice();  //defined in buildDevice.cpp
+
+
+//------------------------------------------------------------------------------
+// Attach the device to any two PhysicalFrames in a model.
+// [Step 2, Task D]
+//------------------------------------------------------------------------------
+void connectDeviceToModel(OpenSim::Device& device, OpenSim::Model& model,
+    const std::string& modelFrameAname, const std::string& modelFrameBname)
+{
+    //TODO: Get references to the "anchor" joints in the device.
+    auto& anchorA = device.updComponent<Joint>("anchorA");
+    auto& anchorB = device.updComponent<Joint>("anchorB");
```

Comment:
At first pass, I blindly wrote these with `getComponent` since the instructions only say to "get references". It becomes clear after the next block why you would need `updComponent`. I think it's fine to leave as is if this is supposed to be a teaching moment.